app/controllers/menu_controller.rb

```diff
@@ -1,19 +1,77 @@
 class MenuController < ApplicationController
-    def menu
-        # display the menu for the current day
-        date = Time.now.strftime("%d/%m/%Y")
-        dishes = TemporaryMenu.get_dishes_by_date(date)
-        @dishes = dishes.to_a
-    end
+    skip_before_action :verify_authenticity_token
     
-    def new_date
-        # byebug
-        # One of our hardworking team member spent one week on this bug 
-        # and we think it is worth commemorating
-        date = params[:date]
-        dishes = TemporaryMenu.get_dishes_by_date(date)
+    #finds the dishes and their recipies for a given meal_type and given day
+    # spits out an array in the format [{name:"kukuname",ingredients:[ingredient:"kukuingredient",portion_size:"kukulargeportion"]}]
+    def find_dishes(meal_type,day)
+        dishes_arr = Menu.get_dishes_by_day_type(day,meal_type)
+        dishes = []
+        dishes_arr.each do |dish_menu|
+            dish = Dish.where(id:dish_menu.dish_id).first
+            if dish != nil 
```

Comment:
Should not happen, unless we have corrupted the db. Add an else case to warn that if it happens.